src/lib.rs

```diff
@@ -0,0 +1,408 @@
+#![forbid(unsafe_code)]
+
+mod baseline;
+mod check_release;
+mod config;
+mod dump;
+mod manifest;
+mod query;
+mod templating;
+mod util;
+
+use cargo_metadata::PackageId;
+pub use config::*;
+pub use query::*;
+
+use check_release::run_check_release;
+use trustfall_rustdoc::{load_rustdoc, VersionedCrate};
+
+use dump::RustDocCommand;
+use itertools::Itertools;
+use semver::Version;
+use std::collections::HashSet;
+use std::path::{Path, PathBuf};
+
+/// Test a release for semver violations.
+pub struct Check {
+    /// Which packages to analyze.
+    scope: Scope,
+    current: Rustdoc,
+    baseline: Rustdoc,
+    log_level: Option<log::Level>,
```

Comment:
Is this different than `GlobalConfig.log_level`? I'm confused by the duplication.

Reply:
I decided not to expose GlobalConfig to the user. I'm only exposing the log level.
That's because if the user wants to use `cargo-semver-checks` as a library, I imagine `cargo-semver-checks` won't write to stdout or stderr in the future. So it doesn't make sense to allow the user to customize stdout and stderr.